src/skai/model/xmanager_external_metric_logger.py

```diff
@@ -1,15 +1,36 @@
 """Keras callback for logging metrics to XManager."""
+from typing import Optional, Union
+from skai.model import log_metrics_callback
 
-from skai.model.log_metrics_callback import MetricLogger
+import tensorflow as tf
 from xmanager.vizier.vizier_cloud import vizier_worker
 
 
-class XManagerMetricLogger(MetricLogger):
+class XManagerMetricLogger(log_metrics_callback.MetricLogger):
   """Class for logging metrics to XManager."""
 
-  def __init__(self, trial_name: str = None) -> None:
+  def __init__(self, trial_name: str = None, output_dir: str | None) -> None:
+    """Constructor for XManagerMetricLogger.
+
+    Args:
+      trial_name: A string containing the Vizier trial name. None if 
+        running on local machine.
+      output_dir: A string of the output directory path.
+    """
     self.trial_name = trial_name
-    self.worker = vizier_worker.VizierWorker(trial_name)
+    if trial_name:
+      self.worker = vizier_worker.VizierWorker(trial_name)
+    else:  # Local run. Write to Tensorboard.
+      self._train_summary_writer = (
+          tf.summary.create_file_writer(
+              os.path.join(output_dir, 'tensorboard', 'train')
```

Comment:
We need to import os library before use.

Reply:
Thanks, done!